packages/caver-account/src/accountKey/accountKeyPublic.js

```diff
@@ -0,0 +1,91 @@
+/*
+    Copyright 2020 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const RLP = require('eth-lib/lib/rlp')
+const utils = require('../../../caver-utils')
+const { ACCOUNT_KEY_TAG } = require('./accountKeyHelper')
+
+/**
+ * Representing an AccountKeyPublic.
+ * @class
+ */
+class AccountKeyPublic {
+    /**
+     * Decodes an RLP-encoded AccountKeyPublic string.
+     * @param {string} rlpEncodedKey - An RLP-encoded AccountKeyPublic string.
+     * @return {AccountKeyPublic}
+     */
+    static decode(rlpEncodedKey) {
+        rlpEncodedKey = utils.addHexPrefix(rlpEncodedKey)
+        if (!rlpEncodedKey.startsWith(ACCOUNT_KEY_TAG.ACCOUNT_KEY_PUBLIC_TAG))
+            throw new Error(`Cannot decode to AccountKeyPublic: ${rlpEncodedKey}`)
+
+        const publicKey = RLP.decode(`0x${rlpEncodedKey.slice(ACCOUNT_KEY_TAG.ACCOUNT_KEY_PUBLIC_TAG.length)}`)
+        return new AccountKeyPublic(publicKey)
+    }
+
+    static fromXYPoint(x, y) {
```

Comment:
Don't we need to check the validity of x and y? the values of x and y should be on the curve.

Reply:
I modified isValidPublicKey function in utils to check on curve.
Also in constructor (AccountKeyPublic and WeightedPublicKey), using that function